vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-integration-tests/src/main/java/com/vaadin/flow/component/spreadsheet/tests/fixtures/SimpleCustomEditorFixture.java

```diff
@@ -1,57 +1,190 @@
-    //
-    // @Override
-    // public Component getCustomComponentForCell(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
-    // return null;
-    // }
-    //
-    // @Override
-    // public Component getCustomEditorForCell(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
-    // if (cell != null) {
-    // // textField.focus();
-    // return textField;
-    // }
-    // return null;
-    // }
-    //
-    // @Override
-    // public void onCustomEditorDisplayed(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet,
-    // Component customEditor) {
-    // if (cell != null) {
-    // String cellValue = spreadsheet.getCellValue(cell);
-    // textField.setValue(cellValue);
-    // }
-    // }
-    // }
+    @Override
+    public void onCustomEditorDisplayed(Cell cell, int rowIndex,
+            int columnIndex, Spreadsheet spreadsheet, Sheet sheet,
+            Component customEditor) {
+        // TODO revisar - what should be done here?
+    }
+
+    private Component getCustomEditor(EditorType editorType, int rowIndex,
+            int columnIndex, Spreadsheet spreadsheet) {
+        if (editorType != null) {
+            switch (editorType) {
+            case TEXT_FIELD:
+                if (textField == null) {
+                    initTextField();
+                    textField.addValueChangeListener(e -> spreadsheet
+                            .refreshCells(spreadsheet.createCell(rowIndex,
+                                    columnIndex, e.getValue())));
+                }
+                return textField;
+            case CHECKBOX:
+                if (checkbox == null) {
+                    initCheckbox();
+                    checkbox.addValueChangeListener(e -> spreadsheet
+                            .refreshCells(spreadsheet.createCell(rowIndex,
+                                    columnIndex, e.getValue())));
+                }
+                return checkbox;
+            case DATE_PICKER:
+                if (datePicker == null) {
+                    initDatePicker();
+                    datePicker.addValueChangeListener(e -> spreadsheet
+                            .refreshCells(spreadsheet.createCell(rowIndex,
+                                    columnIndex, e.getValue().format(
+                                            DateTimeFormatter.ISO_DATE))));
+                }
+                return datePicker;
+            case TEXT_AREA:
+                if (textArea == null) {
+                    initTextArea();
+                    textArea.addValueChangeListener(e -> spreadsheet
+                            .refreshCells(spreadsheet.createCell(rowIndex,
+                                    columnIndex, e.getValue())));
+                }
+                return textArea;
+            case COMBO_BOX:
+                // TODO revisar - js exception in mouseover
```

Comment:
This is covered in https://github.com/vaadin/flow-components/pull/4330#discussion_r1045797756

Reply:
Removed TODO comment